ocs_ci/ocs/platform_nodes.py

```diff
@@ -1513,66 +1525,113 @@ def add_node(self):
-                    power_on=True,
-                    **data,
-                )
-            logger.info("Sleeping for 120 sec to settle down the VMs")
-            time.sleep(120)
-
-            # set hostname
-            for node_name in new_nodes_names:
-                for ip in TimeoutSampler(
-                    600,
-                    60,
-                    self.vsphere.find_ip_by_vm,
-                    node_name,
-                    self.datacenter,
-                    config.ENV_DATA["vsphere_cluster"],
-                    self.cluster_name,
-                ):
-                    if not ("<unset>" in ip or "127.0.0.1" in ip):
-                        logger.info("setting host name")
-                        self.wait_for_connection_and_set_host_name(ip, node_name)
-                        break
+            if use_terraform:
+                self.add_nodes_with_terraform()
+            else:
+                self.add_nodes_without_terraform()
 
+            # give some time to settle down the newly added nodes
             time.sleep(self.wait_time)
 
+            # approve pending CSRs
             if constants.CSR_BOOTSTRAPPER_NODE in existing_csr_data:
                 nodes_approve_csr_num = pre_count_csr + self.compute_count
             else:
                 nodes_approve_csr_num = pre_count_csr + self.compute_count + 1
 
             wait_for_all_nodes_csr_and_approve(expected_node_num=nodes_approve_csr_num)
 
+    def add_nodes_with_terraform(self):
+        """
+        Add nodes using terraform
+        """
+        terraform_state_file = os.path.join(
+            self.terraform_data_dir, "terraform.tfstate"
+        )
+        ips_before_adding_nodes = get_module_ip(
+            terraform_state_file, constants.COMPUTE_MODULE
+        )
+        logger.debug(f"Compute IP's before adding new nodes: {ips_before_adding_nodes}")
+
+        # clone openshift installer
+        clone_openshift_installer()
+        self._update_terraform()
+        self._update_machine_conf()
+
+        # initialize terraform and apply
+        os.chdir(self.terraform_data_dir)
+        self.terraform.initialize()
+        self.terraform.apply(self.terraform_var, module=constants.COMPUTE_MODULE)
+        self.terraform.apply(self.terraform_var, module=constants.COMPUTE_MODULE_VM)
+        os.chdir(self.previous_dir)
+
+        # get the newly added compute IPs
+        ips_after_adding_nodes = get_module_ip(
+            terraform_state_file, constants.COMPUTE_MODULE
+        )
+        logger.debug(f"Compute IP's after adding new nodes: {ips_after_adding_nodes}")
+        new_node_ips = list(set(ips_after_adding_nodes) - set(ips_before_adding_nodes))
+        logger.info(f"Newly added compute IP's: {new_node_ips}")
+
+        # inform load balancer regarding newly added nodes
+        lb = LoadBalancer()
+        lb.update_haproxy_with_nodes(new_node_ips)
+        lb.restart_haproxy()
+
+    def add_nodes_without_terraform(self):
```

Comment:
how about just add_nodes and update the docstring that it uses pyvmomi?

Reply:
i am using add_nodes in https://github.com/red-hat-storage/ocs-ci/pull/3825/files#diff-478e5b871f475f1e312384c0eeb216726f11da23e3d3ac4df80aa3ee5d92e441R1507

Reply:
@vasukulkarni let me know if you have any objections or improvements, I will take up in followup PR if needed

Reply:
@vavuthu sounds good, looks fine since its already used.